OpenRA.Mods.Common/UtilityCommands/ExtractEmmyLuaAPI.cs

```diff
@@ -0,0 +1,358 @@
+
+			Console.WriteLine();
+			WriteManual();
+			Console.WriteLine();
+
+			Console.WriteLine();
+			var actorInits = Game.ModData.ObjectCreator.GetTypesImplementing<ActorInit>()
+				.Where(x => !x.IsAbstract && !x.GetInterfaces().Contains(typeof(ISuppressInitExport)));
+			WriteActorInits(actorInits, out var usedEnums);
+			Console.WriteLine();
+
+			Console.WriteLine();
+			WriteEnums(usedEnums);
+			Console.WriteLine();
+
+			Console.WriteLine();
+			var globalTables = Game.ModData.ObjectCreator.GetTypesImplementing<ScriptGlobal>().OrderBy(t => t.Name);
+			WriteGlobals(globalTables);
+			Console.WriteLine();
+
+			var actorProperties = Game.ModData.ObjectCreator.GetTypesImplementing<ScriptActorProperties>();
+			WriteScriptProperties(typeof(Actor), actorProperties);
+
+			var playerProperties = Game.ModData.ObjectCreator.GetTypesImplementing<ScriptPlayerProperties>();
+			WriteScriptProperties(typeof(Player), playerProperties);
+		}
+
+		static void WriteManual()
+		{
+			Console.WriteLine("--- This function is triggered once, after the map is loaded.");
+			Console.WriteLine("function WorldLoaded() end");
+			Console.WriteLine();
+			Console.WriteLine("--- This function will hit every game tick which by default is every 40 ms.");
+			Console.WriteLine("function Tick() end");
+			Console.WriteLine();
+			Console.WriteLine();
+			Console.WriteLine("--- Base engine types.");
+			Console.WriteLine("---@class cpos");
+			Console.WriteLine("---@field X integer");
+			Console.WriteLine("---@field Y integer");
+			Console.WriteLine();
+			Console.WriteLine("---@class wpos");
+			Console.WriteLine("---@field X integer");
+			Console.WriteLine("---@field Y integer");
+			Console.WriteLine("---@field Z integer");
+			Console.WriteLine();
+			Console.WriteLine("---@class wangle");
+			Console.WriteLine("---@field Angle integer");
+			Console.WriteLine();
+			Console.WriteLine("---@class wdist");
+			Console.WriteLine("---@field Length integer");
+			Console.WriteLine();
+			Console.WriteLine("---@class wvec");
+			Console.WriteLine("---@field X integer");
+			Console.WriteLine("---@field Y integer");
+			Console.WriteLine("---@field Z integer");
+			Console.WriteLine();
+			Console.WriteLine("---@class cvec");
+			Console.WriteLine("---@field X integer");
+			Console.WriteLine("---@field Y integer");
+			Console.WriteLine();
+			Console.WriteLine("---@class color");
+			Console.WriteLine("local color = { };");
+		}
+
+		static void WriteActorInits(IEnumerable<Type> actorInits, out IEnumerable<Type> usedEnums)
+		{
+			var inits = actorInits.Select(x =>
+				(x.Name.Substring(0, x.Name.Length - 4), x.GetConstructors()));
+
+			Console.WriteLine("---A list of ActorInit implementations that can be used by Lua scripts.");
+			Console.WriteLine("---@class initTable");
+
+			var localEnums = new List<Type>();
+			foreach (var (name, constructorInfos) in inits)
+			{
+				var parameters = constructorInfos.Select(ci => ci.GetParameters());
```

Comment:
The LINQ code above can be merged as:
```suggestion
			foreach (var init in actorInits)
			{
				var name = init.Name.Substring(0, init.Name.Length - 4);
				var parameters = init.GetConstructors().Select(ci => ci.GetParameters());
```